src/core/interfaces.ts

```diff
@@ -476,6 +476,29 @@ export namespace CoreChartProps {
   export type I18nStrings = CartesianI18nStrings & PieI18nStrings;
 }
 
+export interface CoreLegendProps {
+  series: Highcharts.SeriesOptionsType[];
```

Comment:
Why does it take HC series?

In the intended use case, a single legend can control multiple charts with different series. E.g.:

```
const lineSeries = [
  { type: 'line', name: 'A', data: [...] },
  { type: 'line', name: 'B', data: [...] }
]

const pieSeries = [
  {
    type: 'pie',
    data: [{ name: 'A', y: 1 }, { name: 'B', y: 2 }] },
]
```

As you can see, the series have a very different structure, yet are represented alike in the legend. Which one should we pass?

I think that instead we should be abstracted away from series and pass some legend items. The items are described by id, label, color, etc. - only the props that are relevant in context of the legend component.